src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -239,7 +239,7 @@ export class EditorPart extends Part implements IEditorPart, IEditorGroupService
 		// while the UI is not yet ready. Clients have to deal with this fact and we have to make sure that the
 		// stacks model gets updated if any of the UI updating fails with an error.
 		const group = this.ensureGroup(position, !options || !options.preserveFocus);
-		const pinned = !this.previewEditors || (options && (options.pinned || typeof options.index === 'number')) || input.isDirty();
+		const pinned = (!this.previewEditors && !(options && options.forcePreview)) || (options && (options.pinned || typeof options.index === 'number')) || input.isDirty();
```

Comment:
When you say quick open, do you mean the control+tab or Open anywhere?  Sorry, Im not up to speed on all the nomenclature.

Won't dirty files already be opened an not previewed, so when they get selected, the editor will just be opened (switched to foreground)?